todo-mvc/src/processes/todoProcesses.ts

```diff
@@ -0,0 +1,142 @@
+import { find, findIndex } from '@dojo/shim/array';
+import { add, replace, remove } from 'le-store/state/operations';
+import {
+	CommandRequest,
+	successResponse,
+	failureResponse
+} from 'le-store/command';
+
+function byId(id: string) {
+	return (todo: any) => id === todo.id;
+}
+
+function throwIfNotOk(response: Response) {
+	if (!response.ok) {
+		throw new Error();
+	}
+	return response;
+}
+
+function byCompleted(completed: boolean) {
+	return (todo: any) => completed === todo.completed;
+}
+
+function updateTodoOperationFactory(get: any, payload: any) {
+	const todos = get('/todos');
+	const todo = find(todos, byId(payload.id));
+	const index = todos.indexOf(todo);
+	return replace(`/todos/${index}`, { ...todo, ...payload });
+}
+
+function addTodoCommand({ get, payload }: CommandRequest) {
+	const todos = get('/todos');
+	return successResponse(add(`/todos/${todos.length}`, payload));
+}
+
+function calculateCountsCommand({ get }: CommandRequest) {
+	const todos = get('/todos');
+	const completedTodos = todos.filter((todo: any) => todo.completed);
+	return successResponse([
+		replace('/activeCount', todos.length - completedTodos.length),
+		replace('/completedCount', completedTodos.length)
+	]);
+}
+
+function toggleAllTodosCommand({ get }: CommandRequest) {
+	const todos = get('/todos');
+	const shouldComplete = !!find(todos, byCompleted(false));
+	const updatedTodos = todos.map((todo: any) => {
+		return { ...todo, completed: shouldComplete };
+	});
+	return successResponse(replace('/todos', updatedTodos));
+}
+
+function clearCompletedCommand({ get }: CommandRequest) {
+	const todos = get('/todos');
+	const activeTodos = todos.filter(byCompleted(false));
+	return successResponse(replace('/todos', activeTodos));
+}
+
+function todoInputCommand({ payload }: CommandRequest) {
+	return successResponse(replace('/currentTodo', payload));
+}
+
+function toggleTodoCommand({ get, payload: [ id, completed ] }: CommandRequest) {
+	return successResponse(updateTodoOperationFactory(get, { id, completed: !completed }));
+}
+
+function editTodoCommand({ get, payload: [ id ] }: CommandRequest) {
+	return successResponse(updateTodoOperationFactory(get, { id, editing: true }));
+}
+
+function saveTodoCommand({ get, payload: [ id, label ] }: CommandRequest) {
+	const todo: any = { id, editing: false };
+	if (label) {
+		todo.label = label;
+	}
+	return successResponse(updateTodoOperationFactory(get, todo));
+}
+
+function clearFailedCommand() {
+	return successResponse(replace('/failed', false));
+}
+
+function postTodoCommand({ get, payload }: CommandRequest) {
+	const fetchOptions = {
+		body: JSON.stringify(payload),
+		method: 'POST',
+		headers: { 'Content-Type': 'application/json' }
+	};
+	return fetch('/todo', fetchOptions)
```

Comment:
If a command has an async requirement then simply return a promise that returns a command response